cmd/podman/containers/ps.go

```diff
@@ -221,7 +221,10 @@ func ps(cmd *cobra.Command, _ []string) error {
 	}
 
 	hdrs, format := createPsOut()
+
+	noHeading, _ := cmd.Flags().GetBool("noheading")
 	if cmd.Flags().Changed("format") {
+		noHeading = !report.HasTable(listOpts.Format)
```

Comment:
I think you need to conflict this if user set noheading, this should not override their setting. I guess you should do this afterwards wrapped with the 

```
        noHeading, _ := cmd.Flags().GetBool("noheading")
	if cmd.Flags().Changed("format") {
                if !cmd.Flags().Changed("noheading") {
		      noHeading = !report.HasTable(listOpts.Format)
                }
...
```